docker/pypi/wmagent/wmagent-docker-build.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+### This script is to be used for building a WMAgent docker imagge based on pypi
+### It depends on a single parameter WMA_TAG, which is to be passed to the basic
+### WMAgent deployment script install.sh at build time through `docker --build-arg`
+
+
+help(){
+    echo -e $1
+    cat <<EOF
+
+The WMAgent docker build script for Docker image creation based on pypi:
+
+Usage: wmagent-docker-build.sh -v <wmagent_tag>
+
+    -v <wmagent_tag>    The WMAgent version/tag to be used for the Docker image creation
+
+Example: ./wmagent-docker-build.sh -v 2.2.0.2
+
+EOF
+}
+
+usage(){
+    help $1
+    exit 1
+}
+
+WMA_TAG=None
+
+### Argument parsing:
+while getopts ":v:h" opt; do
+    case ${opt} in
+        v) WMA_TAG=$OPTARG ;;
+        h) help; exit $? ;;
+        \? )
+            msg="Invalid Option: -$OPTARG"
+            usage "$msg" ;;
+        : )
+            msg="Invalid Option: -$OPTARG requires an argument"
+            usage "$msg" ;;
+    esac
+done
+
+
+
+# NOTE: NO WMA_TAG validation is done in the current script. It is implemented at the install.sh
+
+dockerOpts=" --network=host --progress=plain --build-arg WMA_TAG=$WMA_TAG "
+
+docker build $dockerOpts -t wmagent:$WMA_TAG -t wmagent:latest  .
```

Comment:
I didn't explore this very much, but I was failing to push the new image to CERN registry. I had to rename the image and only then I managed to.
Said that, you might have to update this docker build command to actually define a tag like `registry.cern.ch/cmsweb/wmagent:$WMA_TAG`

Reply:
This is resolved now. 
With my latest commit I introduceed the option `-p` to `wmagent-docker-build.sh`, which triggers a proper tagging and uploading the so built image to `registry.cern.ch`. If not used the build will be local only.

Reply:
Well the `latest` tag for the local image is needed. It achieves  two goals:
* simplifying docker commands for running and maintaining the agent to just `docker <command> wmagent` and avoids the complexity of pointing the full `<repository/image:tag>` tuple every time
* it guaranties we run one and only one agent instance/docker container at the host 

We can skip tagging it as `latest` during the upload step though. That way we will not affect the CI/CD pipeline.

Reply:
providing a second flag to the build wrapper script gives  the expected behavior. It will upload the latest tag to registry.cern.ch only if requested by calling it with:
```
./wmagent-docker-build.sh -v 2.2.1 -p -l
```

The new option is described in the --help message as well. All coming with my next commit